### PR TITLE
Add folder and problems for ADNLPs

### PR DESCRIPTION
**Pull request by tmigot**
Bringing problems in ADNLPModels format #62 . These problems include a `meta` describing the problems.

Instead of using ADNLPModels directly, we could use `Requires` as the breakage tests of `ADNLPModels.jl` would be enough to see breaking releases.

145 problems from the PureJuMP submodule are missing here, and 3 problems here are not in PureJuMP.

### CODE REVIEW

**Comment by dpo** [comment]
> 145 problems from the PureJuMP submodule are missing here, and 3 problems here are not in PureJuMP.

Do you think you could make a list of those problems in an issue?

File: src/ADNLPProblems/ADNLPProblems.jl [hunk 1]
```diff
@@ -0,0 +1,18 @@
+module ADNLPProblems
+
+using ADNLPModels, LinearAlgebra
+import ..PureJuMP
```
Comment (dpo) [inline src/ADNLPProblems/ADNLPProblems.jl:4]: Why do we need PureJuMP in here?

File: Project.toml [hunk 2]
```diff
@@ -3,14 +3,18 @@ uuid = "5049e819-d29b-5fba-b941-0eee7e64c1c6"
 version = "0.3.0"
 
 [deps]
+ADNLPModels = "54578032-b7ea-4c30-94aa-7cbd1cce6c9a"
```
Comment (dpo) [inline Project.toml:6]: Let's use Requires.

**Review by dpo — COMMENTED** [review]
Thank you! How are we going to check that the implementation of those problems is correct?

File: src/ADNLPProblems/ADNLPProblems.jl (hunk 1 above)
Comment (tmigot) [reply src/ADNLPProblems/ADNLPProblems.jl]: Sorry for that, I removed it.

File: Project.toml (hunk 2 above)
Comment (tmigot) [reply Project.toml]: > > 145 problems from the PureJuMP submodule are missing here, and 3 problems here are not in PureJuMP.
> 
> Do you think you could make a list of those problems in an issue?

No problem, I will do it as soon as this one is merged (unless you prefer to list to be completed first?)

**Comment by tmigot** [comment]
> > 145 problems from the PureJuMP submodule are missing here, and 3 problems here are not in PureJuMP.
> 
> Do you think you could make a list of those problems in an issue?

No problem, I will do it as soon as this one is merged (unless you prefer to list to be completed first?)

**Comment by tmigot** [comment]
> Thank you! How are we going to check that the implementation of those problems is correct?

I planned a sanity check comparing the NLPModelsJuMP version of the JuMP models, and the ADNLPModels. Similar to `test/compare_cutest.jl`. I think it can be done in a separate PR improving the unit tests.

**Comment by dpo** [comment]
FreeBSD tests Julia 1.0. That should be changed to 1.3.

**Review by dpo — APPROVED** [review]

**Comment by abelsiqueira** [comment]
Thanks

**Pull request merged**